src/python/WMCore/Services/DBS/DBSErrors.py

```diff
@@ -58,3 +63,53 @@ class DBSReaderError(DataMgmtError):
     """
     def __init__(self, msg, **data):
         DataMgmtError.__init__(self, msg, 1002, **data)
+
+
+class DBSError():
+    """
+    DBSError provides generic interface for DBS (Go-based) errors
+    """
+    def __init__(self, ex):
+        try:
+            if hasattr(ex, "body"):
+                # case of DBSClient HTTPError
+                self.data = json.loads(ex.body)[0]
+            else:
+                self.data = json.loads(ex)[0]
+        except Exception as exp:
+            self.data = str(ex)
+
+    def getCode(self):
+        """
+        :return: DBS server error code
+        """
+        if isinstance(self.data, dict):
+            return self.data['error']['code']
+        return self.data
+
+    def getMessage(self):
+        """
+        :return: DBS server error message (consice output, last error in DBS error chain)
+        """
+        if isinstance(self.data, dict):
+            return self.data['error']['message']
+        return self.data
+
+    def getReason(self):
+        """
+        :return: DBS server error reason (expanded message)
+        """
+        if isinstance(self.data, dict):
+            return self.data['error']['reason']
+        return self.data
+
+    def getServerCode(self):
+        """
+        Return appropriate message based on giver HTTPError server code
+        """
+        dbsCodes = "https://github.com/dmwm/dbs2go/blob/master/dbs/errors.go"
+        if hasattr(self.data, "server_code"):
```

Comment:
Isn't `self.data` supposed to be a dictionary object?

Second comment on this block is, isn't the `code` key nested into an `error` dictionary? E.g.:
```
[{"error": {"code": 123, ...
```

Lastly, this method should be returning an integer (the server_code error number). Maybe we return None if we cannot find the error/code in the error message/object?

Reply:
It must have been a miscommunication then, sorry.
What `getServerCode()` method should return is a simple integer value with the error code reported by the DBS server. Regarding the url with the DBS error code definition, I suggested to actually add them as a comment in this class, such that whenever we look into the source code, we know from where they come from and what their meaning is.